webplatformcompat/renderers.py

```diff
@@ -1,244 +1,242 @@
+        linked = self.dict_class()
+        meta = self.dict_class()
+
+        for name, value in data.items():
+            field_extra = fields_extra.get(name, {})
+            link = field_extra.get('link')
+            attr_name = field_extra.get('name', name)
+            if link:
+                link_pattern, link_id = self.convert_link(
+                    name, value, field_extra, main_resource, request)
+                if link_pattern:
+                    link_ids[attr_name] = link_id
+                    pattern_name = "%s.%s" % (main_resource, attr_name)
+                    link_patterns[pattern_name] = link_pattern
+                else:
+                    attributes[attr_name] = link_id
+            elif name == '_view_extra':
+                meta.update(value.pop('meta', {}))
+                for ve_key, ve_list in value.items():
+                    converted_list, resource = self.convert_list(
+                        ve_list, request)
+                    assert resource == ve_key, (
+                        "%s != %s" % (resource, ve_key))
+                    linked[ve_key] = converted_list[resource]
+                    link_patterns.update(converted_list.get('links', {}))
+            else:
+                attributes[attr_name] = value
+
+        converted = self.dict_class(((main_resource, attributes),))
+        if link_ids:
+            converted[main_resource]['links'] = link_ids
+        if link_patterns:
+            converted['links'] = link_patterns
+        if linked:
+            converted['linked'] = linked
+        if meta:
+            converted['meta'] = meta
+        return converted
+
+    def convert_paginated(self, data, request):
+        """Convert paginated data to JSON API RC1.
+
+        The pagination data is moved to the "meta" key, and the
+        paginated "results" are split into resources and link data.
+        """
+        keys = set(data.keys())
+        is_paginated = keys >= set(('count', 'next', 'previous', 'results'))
+        converted = self.dict_class()
+
+        assert is_paginated
+        converted, resource = self.convert_list(data['results'], request)
+        pagination = self.dict_class((
+            ('previous', data['previous']),
+            ('next', data['next']),
+            ('count', data['count']),
+        ))
+        converted['meta'] = self.dict_class((
+            ('pagination', self.dict_class((
+                (resource, pagination),
+            ))),
+        ))
+        return converted
+
+    def convert_list(self, return_list, request):
+        """Convert a ReturnList to JSON API RC1 format.
+
+        The list data is moved under the resource name, and the link metadata
+        is collected under the "links" key, using field data from the
+        ReturnList's attached ListSerializer, and using the request to build
+        full URLs.
+        """
+        assert hasattr(return_list, 'serializer'), "Must be ReturnList"
```

Comment:
or something close enough to a ReturnList. I think I got burned by the DRF 2.x to 3.x transition when this changes. ``isinstance()`` will be clearer, though, so I'll make the change.